src/video/x11/SDL_x11dyn.c

```diff
@@ -32,52 +32,78 @@
 
 #ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC
 
-typedef struct
-{
-    SDL_SharedObject *lib;
-    const char *libname;
-} x11dynlib;
-
-#ifndef SDL_VIDEO_DRIVER_X11_DYNAMIC_XEXT
-#define SDL_VIDEO_DRIVER_X11_DYNAMIC_XEXT NULL
+#ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC_XEXT
+#define SDL_X11_DYNAMIC_XEXT_APPEND ,SDL_VIDEO_DRIVER_X11_DYNAMIC_XEXT
+#else
+#define SDL_X11_DYNAMIC_XEXT_APPEND
 #endif
-#ifndef SDL_VIDEO_DRIVER_X11_DYNAMIC_XCURSOR
-#define SDL_VIDEO_DRIVER_X11_DYNAMIC_XCURSOR NULL
+
+#ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC_XCURSOR
+#define SDL_X11_DYNAMIC_XCURSOR_APPEND ,SDL_VIDEO_DRIVER_X11_DYNAMIC_XCURSOR
+#else
+#define SDL_X11_DYNAMIC_XCURSOR_APPEND
 #endif
-#ifndef SDL_VIDEO_DRIVER_X11_DYNAMIC_XINPUT2
-#define SDL_VIDEO_DRIVER_X11_DYNAMIC_XINPUT2 NULL
+
+#ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC_XINPUT2
+#define SDL_X11_DYNAMIC_XINPUT2_APPEND ,SDL_VIDEO_DRIVER_X11_DYNAMIC_XINPUT2
+#else
+#define SDL_X11_DYNAMIC_XINPUT2_APPEND
 #endif
-#ifndef SDL_VIDEO_DRIVER_X11_DYNAMIC_XFIXES
-#define SDL_VIDEO_DRIVER_X11_DYNAMIC_XFIXES NULL
+
+#ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC_XFIXES
+#define SDL_X11_DYNAMIC_XFIXES_APPEND ,SDL_VIDEO_DRIVER_X11_DYNAMIC_XFIXES
+#else
+#define SDL_X11_DYNAMIC_XFIXES_APPEND
 #endif
-#ifndef SDL_VIDEO_DRIVER_X11_DYNAMIC_XRANDR
-#define SDL_VIDEO_DRIVER_X11_DYNAMIC_XRANDR NULL
+
+#ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC_XRANDR
+#define SDL_X11_DYNAMIC_XRANDR_APPEND ,SDL_VIDEO_DRIVER_X11_DYNAMIC_XRANDR
+#else
+#define SDL_X11_DYNAMIC_XRANDR_APPEND
 #endif
-#ifndef SDL_VIDEO_DRIVER_X11_DYNAMIC_XSS
-#define SDL_VIDEO_DRIVER_X11_DYNAMIC_XSS NULL
+
+#ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC_XSS
+#define SDL_X11_DYNAMIC_XSS_APPEND ,SDL_VIDEO_DRIVER_X11_DYNAMIC_XSS
+#else
+#define SDL_X11_DYNAMIC_XSS_APPEND
 #endif
-#ifndef SDL_VIDEO_DRIVER_X11_DYNAMIC_XTEST
-#define SDL_VIDEO_DRIVER_X11_DYNAMIC_XTEST NULL
+
+#ifdef SDL_VIDEO_DRIVER_X11_DYNAMIC_XTEST
+#define SDL_X11_DYNAMIC_XTEST_APPEND ,SDL_VIDEO_DRIVER_X11_DYNAMIC_XTEST
+#else
+#define SDL_X11_DYNAMIC_XTEST_APPEND
 #endif
 
-static x11dynlib x11libs[] = {
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC },
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC_XEXT },
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC_XCURSOR },
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC_XINPUT2 },
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC_XFIXES },
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC_XRANDR },
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC_XSS },
-    { NULL, SDL_VIDEO_DRIVER_X11_DYNAMIC_XTEST }
+#define SDL_X11_DYNAMIC_LIBS \
+    SDL_VIDEO_DRIVER_X11_DYNAMIC \
+    SDL_X11_DYNAMIC_XEXT_APPEND \
+    SDL_X11_DYNAMIC_XCURSOR_APPEND \
+    SDL_X11_DYNAMIC_XINPUT2_APPEND \
+    SDL_X11_DYNAMIC_XFIXES_APPEND \
+    SDL_X11_DYNAMIC_XRANDR_APPEND \
+    SDL_X11_DYNAMIC_XSS_APPEND \
+    SDL_X11_DYNAMIC_XTEST_APPEND
+
+SDL_ELF_NOTE_DLOPEN(
+    "x11",
+    "Support for video through X11 backend",
+    SDL_ELF_NOTE_DLOPEN_PRIORITY_REQUIRED,
```

Comment:
I think these should be `RECOMMENDED`. You don't strictly need X11 libraries if you are actually running on native Wayland, and if packagers are OK with having a hard dependency then they can turn off `SDL_VIDEO_DRIVER_X11_DYNAMIC` (as we do in Debian).